### ISSUE TITLE
Add HTTP/3 Support to zTunnel

### ISSUE THREAD

**Issue by stevenctl** [istio/ztunnel#423]
HTTP/3 and QUIC may have benefits for TCP proxying, and are required before we can implement #148.

The first step is choosing a lib. For now, looking at https://github.com/hyperium/h3. Seems that support in https://github.com/hyperium/hyper is further off. 

See this [doc](https://docs.google.com/document/d/1MwJlCl0ndhS5ZXSuSTdQUqCb-IGu23m3h7AcnPeHFm0/edit?resourcekey=0-vJ778OjdknOd_QskWPpndg#) for more info. 

- [ ] Implement CONNECT in `h3` lib
- [ ] Design for how to expose h2 vs h3 
- [ ] Abstract some of the HTTP code in zTunnel to allow bringing in the lib
- [ ] Add h3 support properly

**Comment by PiotrSikora** [comment]
> HTTP/3 and QUIC may have benefits for TCP proxying, and are required before we can implement #148.

They are not strictly required for it, since you can proxy UDP over HTTP/2 without dropping any packets.

Also, this seems like a duplicate of https://github.com/istio/ztunnel/issues/150?
